src/commands/targeting/enable.ts

```diff
@@ -23,47 +28,26 @@ export default class EnableTargeting extends Base {
 
         await this.requireProject()
 
-        let responses
-
-        const feature = args['feature']
-        const environment = args['environment']
-
-        if (flags.headless && (!feature || !environment)) {
-            throw new Error('In headless mode, both the feature and environment are required')
-        }
-
-        if (feature) {
-            responses = { featureKey: feature }
-        } else {
-            const userSelectedFeature = await inquirer.prompt(
-                [featurePrompt],
-                {
-                    token: this.authToken,
-                    projectKey: this.projectKey
-                }
-            )
-            responses = { featureKey: userSelectedFeature.feature }
-        }
-
-        if (environment) {
-            responses = { ...responses, environmentKey: environment }
-        } else {
-            const userSelectedEnv = await inquirer.prompt<EnvironmentPromptResult>(
-                [environmentPrompt],
-                {
-                    token: this.authToken,
-                    projectKey: this.projectKey
-                }
-            )
-            responses = { ...responses, environmentKey: userSelectedEnv.environment._id }
-        }
-
-        const enableTargetingForFeatureAndEnvironment = await enableTargeting(
+        const responses = await getResponsesForEnableDisableTargeting(
+            this.authToken, 
+            this.projectKey, 
+            args, 
+            flags,
+        )
+        const updatedTargeting = await enableTargeting(
             this.authToken,
             this.projectKey,
-            responses.featureKey,
-            responses.environmentKey
-        )
-        return this.writer.showResults(enableTargetingForFeatureAndEnvironment)
+            responses.featureKey as string,
+            responses.environmentKey as string
+        ) 
+    
+        if (flags.headless) {
+            this.writer.showResults(getResponsesForEnableDisableTargeting)
```

Comment:
shouldn't this take `updatedTargeting` rather than `getResponsesForEnableDisableTargeting`?

Reply:
oops yeah